srcpkgs/xen/files/xenconsoled/check

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+exec > /dev/null
+exec 2>&1
+
+# Check for the faux xenconsoled first:
+pgrep -f '^shim-xenconsoled' && exit 0
+
+# There's not really a good way to check for xenconsoled being up.
+# The best I can come up with is to look at the open FDs, and if
+# hypervisor.log is open, then xenconsoled is almost certainly started
+# and ready to use.
+PID="$(pidof xenconsoled)" || exit 1
+expr "x${PID}" ':' 'x[0-9]\+$' > /dev/null || exit 1
+ls -l "/proc/${PID}/fd" 2>/dev/null |grep -q hypervisor.log
```

Comment:
Consider changing this to `lsof -p ${PID:-0} | grep -q hypervisor.log` or rolling this and the PID= line into a single command. It still isn't perfect but should be less fiddly.

Reply:
If this check really matters, and there's no reliable way to actually determine whether the service is running, might as well just use `chpst -L` in the run script to acquire a lock on some file like `/run/xen/xenconsoled.lock`. Then the check script can just try to acquire that lock and, if it succeeds, you know the service is not running.

Yes, this would be a hack. It would also be much less hacky than trying to scrape the process tree.

Reply:
Yeah, a lock works too. I don't particularly like them in shell scripts but `chpst -L lockfile ...` in the run script and `flock -n lockfile` in the check script works as well.